src/render/draw_debug.js

```diff
@@ -9,15 +9,60 @@ import SegmentVector from '../data/segment';
 import DepthMode from '../gl/depth_mode';
 import StencilMode from '../gl/stencil_mode';
 import CullFaceMode from '../gl/cull_face_mode';
-import { debugUniformValues } from './program/debug_program';
+import ColorMode from '../gl/color_mode';
+import { debugUniformValues, debugSSRectUniformValues } from './program/debug_program';
 import Color from '../style-spec/util/color';
+import browser from '../util/browser';
 
 import type Painter from './painter';
 import type SourceCache from '../source/source_cache';
 import type {OverscaledTileID} from '../source/tile_id';
 
 export default drawDebug;
 
+const topColor = new Color(1, 0, 0, 0.4);
```

Comment:
Does draw_debug.js go to production code?
I'm not sure how useful is to including additional shader and this code for rendering different colors or padding values.
For manual debugging, propose to add additional debug/***.html page (doesn't go to production code AFAIK) with overlays (in the way the customers are supposed to use it)...

> - [x] manually test the debug page

and in that that page would be used for this step.

Please doublecheck my opinion with other reviewers - I might be missing something...

Reply:
Thanks for adding a debug view. They can be pretty useful for understanding things. If the size and complexity of the shader another approach could be to use a gl scissor and clear to draw a line at each edge.

```js
    gl.enable(gl.SCISSOR_TEST);
    gl.scissor(x, y, width, height);
    context.clear({ color });
    gl.disable(gl.SCISSOR_TEST);
```